### ISSUE TITLE
FeatureSpace.SortSamplesByGroundTruth() is not correct when inplace=False

### ISSUE THREAD

**Issue by igg** [wnd-charm/wnd-charm#100]
FeatureSpace.SortSamplesByGroundTruth() is sorting on _contiguous_ground_truth_values even if they contain None values.  This results in Splits not functioning properly when being done by ShuffleSplit:

    >>> fs = FeatureSpace.NewFromDirectory('train-test', long=True, quiet = True)
    >>> fs2 = fs.SortSamplesByGroundTruth(inplace=False)
    >>> print ('fs2._contiguous_ground_truth_labels: {}'.format(fs2._contiguous_ground_truth_labels))
    fs2._contiguous_ground_truth_labels: ['0', '1', '0', '0', '1', '0', '0', '1', '0', '0', '1', '0', '0', '1', '0', '0', '0', '1', '0', '0', '1', '0', '0', '1', '0', '0', '0', '1', '1', '1']

This works if inplace=True, but inplace=False results in the sort not working.
If a Split is done, it will be messed up:

    >>> train2,test2 = fs2.Split (.8,.2)
    SPLIT FEATURE SPACE INTO TRAINING SET:  <FeatureSpace "train-test (sorted) (sorted) (subset) (sorted)" n_features=2919 n_total_samples=16 n_samples_per_group=1 n_classes=2 samples_per_class=("0": 10, "1": 6)>
    TEST SET:  <FeatureSpace "train-test (sorted) (sorted) (subset) (sorted)" n_features=2919 n_total_samples=6 n_samples_per_group=1 n_classes=2 samples_per_class=("0": 3, "1": 3)>

**Comment by igg** [comment]
Should really set the _contiguous_ground_truth_values list when we check for numeric classes.  No reason to keep them as a list of Nones after the check.
Should also check for None in _contiguous_ground_truth_values, and sort by labels if there are any Nones in there
